arcaflow_plugin_fio/fio_schema.py

```diff
@@ -58,108 +58,357 @@ def is_sync(self) -> bool:
+        validation.min(1),
+        schema.name("Concurrent Open Files"),
+        schema.description(
+            "Number of files to keep open at the same time. Defaults to the same as "
+            "nrfiles, but can be set smaller to limit the number simultaneous opens."
+        ),
+    ] = None
+    create_on_open: typing.Annotated[
+        typing.Optional[int],
+        validation.min(0),
+        validation.max(1),
+        schema.name("Create Files on Open"),
+        schema.description(
+            "Don't pre-create files but allow the job's open() to create a file when "
+            "it's time to do I/O."
+        ),
+    ] = None
+    pre_read: typing.Annotated[
+        typing.Optional[int],
+        validation.min(0),
+        validation.max(1),
+        schema.name("Pre-Read Files"),
+        schema.description(
+            "Files will be pre-read into memory before starting the given I/O "
+            "operation. This will also clear the invalidate flag, since it is "
+            "pointless to pre-read and then drop the cache. This will only work for "
+            "I/O engines that are seek-able, since they allow you to read the same "
+            "data multiple times. Thus it will not work on non-seekable I/O engines "
+            "(e.g. network, splice)."
+        ),
+    ] = None
+    unlink: typing.Annotated[
+        typing.Optional[int],
+        validation.min(0),
+        validation.max(1),
+        schema.name("Unlink Files"),
+        schema.description(
+            "Unlink the job files when done. Not the default, as repeated runs of that "
+            "job would then waste time recreating the file set again and again."
+        ),
+    ] = None
+    unlink_each_loop: typing.Annotated[
+        typing.Optional[int],
+        validation.min(0),
+        validation.max(1),
+        schema.name("Unlink Files Each Loop"),
+        schema.description("Unlink job files after each iteration or loop."),
+    ] = None
+
+    # I/O Type
+    direct: typing.Annotated[
+        typing.Optional[int],
+        validation.min(0),
+        validation.max(1),
+        schema.name("Direct I/O"),
+        schema.description(
+            "Use non-buffered I/O. This is usually O_DIRECT. Note that OpenBSD and ZFS "
+            "on Solaris don't support direct I/O. On Windows the synchronous ioengines "
+            "don't support direct I/O."
+        ),
+    ] = None
     buffered: typing.Annotated[
         Optional[int],
         validation.min(0),
         validation.max(1),
-    ] = field(
-        default=1,
-        metadata={
-            "name": "Buffered",
-            "description": "Use buffered IO if True, else use direct IO.",
-        },
-    )
-    atomic: typing.Annotated[
-        Optional[int],
+        schema.name("Buffered"),
+        schema.description(
+            "Use buffered I/O. This is the opposite of the direct option."
+        ),
+    ] = None
+    readwrite: typing.Annotated[
```

Comment:
I find that variable name a little confusing. Is that the one that FIO uses?

Reply:
Yes, I agree it is confusing, bu this is what fio itself uses, unfortunately. This is part of why I was asking in another thread about our ability to use aliases for schema object names.